sequencer/src/persistence/fs.rs

```diff
@@ -893,14 +934,322 @@ impl SequencerPersistence for Persistence {
+
+        let decided_leaf_path = inner.decided_leaf_path();
+        if !decided_leaf_path.is_dir() {
+            return Ok(());
+        }
+
+        tracing::warn!("migrating decided leaves..");
+        for entry in fs::read_dir(decided_leaf_path)? {
+            let entry = entry?;
+            let path = entry.path();
+
+            let Some(file) = path.file_stem().and_then(|n| n.to_str()) else {
+                continue;
+            };
+            let Ok(view) = file.parse::<u64>() else {
+                continue;
+            };
+
+            let bytes =
+                fs::read(&path).context(format!("reading decided leaf {}", path.display()))?;
+            let (leaf, qc) = bincode::deserialize::<(Leaf, QuorumCertificate<SeqTypes>)>(&bytes)
+                .context(format!("parsing decided leaf {}", path.display()))?;
+
+            let leaf2: Leaf2 = leaf.into();
+            let qc2 = qc.to_qc2();
+
+            let file_path = decided_leaf2_path
+                .join(view.to_string())
+                .with_extension("txt");
+
+            inner.replace(
+                &file_path,
+                |_| {
+                    tracing::warn!(view, "duplicate decided leaf");
+                    Ok(false)
+                },
+                |mut file| {
+                    let bytes = bincode::serialize(&(&leaf2.clone(), qc2))?;
+                    file.write_all(&bytes)?;
+                    Ok(())
+                },
+            )?;
+        }
+
+        inner.migrated.insert("anchor_leaf".to_string());
+        inner.update_migration()?;
+        tracing::warn!("successfully migrated decided leaves");
+        Ok(())
+    }
+    async fn migrate_da_proposals(&self) -> anyhow::Result<()> {
+        let mut inner = self.inner.write().await;
+
+        if inner.migrated.contains("da_proposal") {
+            tracing::info!("da proposals already migrated");
+            return Ok(());
+        }
+
+        let da2_path = inner.da2_dir_path();
+
+        fs::create_dir_all(da2_path.clone()).context("failed to create da proposals 2 dir")?;
+
+        let da_dir = inner.da_dir_path();
+        if !da_dir.is_dir() {
+            return Ok(());
+        }
+
+        tracing::warn!("migrating da proposals..");
+
+        for entry in fs::read_dir(da_dir)? {
+            let entry = entry?;
+            let path = entry.path();
+
+            let Some(file) = path.file_stem().and_then(|n| n.to_str()) else {
+                continue;
+            };
+            let Ok(view) = file.parse::<u64>() else {
+                continue;
+            };
+
+            let bytes =
+                fs::read(&path).context(format!("reading da proposal {}", path.display()))?;
+            let proposal = bincode::deserialize::<Proposal<SeqTypes, DaProposal<SeqTypes>>>(&bytes)
+                .context(format!("parsing da proposal {}", path.display()))?;
+
+            let file_path = da2_path.join(view.to_string()).with_extension("txt");
```

Comment:
I think in these functions we could be a bit more explicit with naming to make it easier to read. As in for example `old_path` `new_path`, `old_file`, `new_file` or something like that, instead of `path`, `file`, `file_path`, `file`.

Reply:
done in https://github.com/EspressoSystems/espresso-sequencer/pull/2478/commits/0ea3d2b4678dfb32f53dc1255ee83eb78e9aab52